src/util.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+export function _sign(x: number): number {
```

Comment:
* I'd like to use a more informative name for this module than `util`, in part to avoid it becoming a catch-all file that accumulates unrelated functions.
* My understanding of Typescript modules/exports/visibility is pretty weak. Can we make this package-visible by using `export` (as you've done here) and then simply not exposing it from the `Ion` module that end users will interact with? If so, can we avoid the underscore prefix?

Reply:
* please suggest a more informative name;  numberutil?  mathutil?  or we can watch how util evolves and refactor if it becomes a problem;  copyright date is 2012 and this appears to be the first cross-module util function....
* at the end of the day, it all becomes JavaScript, which has no protections afaik, so _everything_ is available to the caller;  and there's nothing to stop someone from importing 'util';  in the absence of such protections, i think the '_'  prefix is actually the best option.  E.g., the following TypeScript:
```typescript
import { _sign } from "./util";
_sign(5);
```
transpiles to (commonjs):
```javascript
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const util_1 = require("./util");
util_1._sign(5);
```
or (AMD):
```javascript
define(["require", "exports", "./util"], function (require, exports, util_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    util_1._sign(5);
});
```

Reply:
Javascript allows for extending the `Number` class by adding a function directly to it's prototype. I wonder if this method should live directly on `Number`s, allowing for:
```
2_048.sign()
```
I don't know whether that's considered poor style, however.

> or we can watch how util evolves and refactor if it becomes a problem

I'm fine with this.

Reply:
> at the end of the day, it all becomes JavaScript, which has no protections afaik, so everything is available to the caller; and there's nothing to stop someone from importing 'util'; in the absence of such protections, i think the '_' prefix is actually the best option.

Using a recent build of the `development` branch, I can run:
```
var ion = require('./dist/commonjs/es6/Ion.js')
  // or
var ion = require('./dist/browser/js/ion-bundle.js')
```
Either of which gives me access to only those functions/classes that were exported from `Ion.ts`. I can't get a handle to the contents of `IonSpan` using my `ion` variable declared above, for example.

As a developer I might be able to jump through some hoops to access that, but it's not ergonomically available. I would expect that we wouldn't export the contents of `util` or the module itself, and so am wondering if we really need the underscore to convince developers not to use it.

At any rate, this doesn't need to block your PR.

Reply:
Right, I've since learned that the notion of a module does offer some protection.  I think the question becomes whether we care whether consumers do something like:
```
var util = require('./dist/commonjs/es6/util.js')
```

Reply:
> Javascript allows for extending the `Number` class by adding a function directly to it's prototype.

Not sure it's warranted in this case, and it raises other questions:
* if another library also defines Number.sign(), which impl wins?
* feels like our API would be creeping outside it's boundary;  sounds appropriate for something like a `toIon()` method, but I'm not sure `sign()` should be any part of this library's API